### ISSUE TITLE
[Dev Aks] Handheld weapon feedback and blueprints!

### ISSUE THREAD

**Issue by tsunamayo** [tsunamayo/Starship-EVO#2242]
So what is you feedback for handheld weapons? What could we add? I still need to fix a few bugs and do something for children entity.
Also I am very much looking forward to see a few of your builds! Thanks

**Comment by Aiwii** [comment]
![image](https://user-images.githubusercontent.com/53021785/83084258-ad5d9380-a045-11ea-849c-7c4441efe26b.png)
[WX BOOMSTICK AND MAKRA.zip](https://github.com/tsunamayo/Starship-EVO/files/4692504/WX.BOOMSTICK.AND.MAKRA.zip)

Some clipping on the MAKRA no clipping on the BOOMSTICK IIRC

**Comment by Dwarf-LordPangolin** [comment]
First, the system is brilliant; I've never seen a game in this genre that allows for this degree of creativity in player weapons.

The only real feedback I have is that maybe the brick scaling should be reduced a little more? Not a ton, just another 10% maybe (but not the handle, obviously ;) ). And maybe another variant of the handle brick that's more gun-handle shaped could be done at some point in the future?

I've only got one completed gun so far, but I've got plans for many more!
![20200527164232_1](https://user-images.githubusercontent.com/33041786/83084332-3bd61300-a04e-11ea-9bac-610a37ea0aa6.jpg)
[Eisenstern Armaments SAW-33.zip](https://github.com/tsunamayo/Starship-EVO/files/4692505/Eisenstern.Armaments.SAW-33.zip)

**Comment by tsunamayo** [comment]
Yes I will experiment with a smaller scale. 50% was okay too - but like I said I think we will use them on mini mech also, so they have to stay on the bulky side...

**Comment by Alchemichai** [comment]
There was recently a suggestion[](https://github.com/tsunamayo/Starship-EVO/issues/2241)  to have weapons scalable so users could have the weapons whatever size they want (within reason). I suppose then weapons would also need to take scale into account for damage stats.

**Comment by ZachZent** [comment]
I'm sure this one is planned, but using the weapons right now requires an energy brick. Maybe in the future there can be a some sort of ammo holder brick. With these you can insert ammo depending on the weapon. Bullets for future kinetic weapons, "batteries" for lasers or beams.

Keep the heat mechanics for weapons, but I'd make a separate system for hand weapons versus ships. For example it would go up to 100% before overheating (not blowing up) rather than 75%. Something combined with ammo this would make a good balance for heavy vs lighter weapons.

The rate of fire really need to be increased. This is way too low to be interesting or engaging. Firing at the rate of a bolt action rifle will never be as fun as "han solo" running down a corridor firing as fast as possible. Don't worry about the gatling gun, it will still be way faster than any semi-auto fire.

https://cdn.discordapp.com/attachments/406664114249793536/715363806439538740/Starship_EVO_2020-05-27_17-39-28.mp4

**Comment by Xenoprophet-Nail** [comment]
Decals, and perhaps a rifle styled grip, so a sniper rifle isn't wielded in one hand.

**Comment by Womble-TC** [comment]
Made a backpack gun that sorta works 
![image](https://user-images.githubusercontent.com/29961381/83088176-ecfa9e80-a08a-11ea-8d99-5c642987f1f3.png)
![image](https://user-images.githubusercontent.com/29961381/83088232-0dc2f400-a08b-11ea-85c6-b2e75b8de1a5.png)
[BackPack Gun.zip](https://github.com/tsunamayo/Starship-EVO/files/4692703/BackPack.Gun.zip)

**Comment by Xenoprophet-Nail** [comment]
ooh, now all we need is hinges for a shoulder cannon, or a manipulator arm

**Comment by Womble-TC** [comment]
wanted the guns to fold up and down, but hinges don't work on guns (this is probably intentional).

**Comment by Xenoprophet-Nail** [comment]
lol I wanted to make one of my turret guns, so it  looked like it was salvaged from a turret, like they do in Halo and other games.

**Comment by tsunamayo** [comment]
I want to support hinge and the like! I will need a builtin menu to set min max rotation value though, going with the normal sequencer brick is not okay for such tiny build.
Amazing idea Womble, now I want to make custom jetpack ;)...
Ho yes with a turret gimbaled on top that would be killer...

**Comment by Xenoprophet-Nail** [comment]
Epic man! I needed this, I had posted about not being able to use some logic anymore, I had a weapon that had a laser that says "target acquired" when someone breaks the beam lol. Good for a sniper rifle in lieu of a scope and excellent for a booby trap in a ship lol

**Comment by hl2shadowslayer** [comment]
I made a dual-wielded pistol, but the rifle animations aren't exactly doing it any favors. Would be nice if we had a way dual-wield one handed weapons.
[Dual Blaster Pistols.zip](https://github.com/tsunamayo/Starship-EVO/files/4692756/Dual.Blaster.Pistols.zip)
![20200526182547_1](https://user-images.githubusercontent.com/28115168/83089150-75a51a80-a04a-11ea-85af-fa7173788cb7.jpg)

**Comment by Xenoprophet-Nail** [comment]
An Autocannon

![20200528023825_1](https://user-images.githubusercontent.com/51257864/83208880-ddc82f00-a124-11ea-904d-1c1f09d13066.jpg)

**Comment by ExodistSKY1** [comment]
I have been busy with things going on and have yet to have my hand at the personal weapons. 

I understand you want these weapons to be used with mini mechs, I am guessing on par with power suits we will hopefully also have smaller blocks for ;) ,    Is there a way you can change the scaling between if a mini mech picks up the weapon or if it is picked up by a player/NPC?

The bulky weapons do look okay, because we have some talented builders, but limit the creativity that this community has to over and short changes the games switch to the new art style. 

I have been keeping up to date and watching the builds on discord, here are some of my suggestions:

    -Add another pistol grip to allow for variation between weapons 
    -Make the current trigger grip a little smaller, it does not look good with the current character hand size 
    -Add a foregrip attachment for the off-hand to know where to be placed when picking up the weapon by either a mech or player (bonus points if you can make two or three variations, Vertical, angled, flat)
    -possibly a scope and a reflex sight

These might seem like small things but with the graphics overhaul, I feel it is worth allowing things to look a little better as opposed to looking kind of out of place. 

I can't wait to get in and try my hand at the weapons and I still have a lot of other testing videos to make and upload for you to see "actual" gameplay.

**Comment by Xenoprophet-Nail** [comment]
well he's looking into a bug with how a second grip isn't correctly rotated where a rifle grip would work so that may get fixed sometime soon.

**Comment by ultrak2k** [comment]
![20200601150028_1](https://user-images.githubusercontent.com/43084360/83374397-a35ddb00-a41f-11ea-8691-b6d077b132f4.jpg)
hey tsuna ya like touhou?

**Comment by ultrak2k** [comment]
There was an issue while building the above - using logic on weapons doesn't work as you cant press F on them
